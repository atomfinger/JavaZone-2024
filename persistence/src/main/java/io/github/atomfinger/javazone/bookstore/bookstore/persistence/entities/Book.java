package io.github.atomfinger.javazone.bookstore.bookstore.persistence.entities;

import jakarta.persistence.*;

import java.util.Date;

@Entity
@Table(name = "books")
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "book_id")
    private Long bookId;

    @Column(name = "title", nullable = false, length = 255)
    private String title;

    @Column(name = "description")
    private String description;

    @Column(name = "isbn", nullable = false, unique = true, length = 13)
    private String isbn;

    @Column(name = "author_name", nullable = false, length = 255)
    private String authorName;

    @Column(name = "page_numbers")
    private Integer pageNumbers;

    @Column(name = "published_date")
    @Temporal(TemporalType.DATE)
    private Date publishedDate;

    @Column(name = "genre", length = 100)
    private String genre;

    public Long getBookId() {
        return bookId;
    }

    public void setBookId(Long bookId) {
        this.bookId = bookId;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getIsbn() {
        return isbn;
    }

    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }

    public String getAuthorName() {
        return authorName;
    }

    public void setAuthorName(String authorName) {
        this.authorName = authorName;
    }

    public Integer getPageNumbers() {
        return pageNumbers;
    }

    public void setPageNumbers(Integer pageNumbers) {
        this.pageNumbers = pageNumbers;
    }

    public Date getPublishedDate() {
        return publishedDate;
    }

    public void setPublishedDate(Date publishedDate) {
        this.publishedDate = publishedDate;
    }

    public String getGenre() {
        return genre;
    }

    public void setGenre(String genre) {
        this.genre = genre;
    }
}
