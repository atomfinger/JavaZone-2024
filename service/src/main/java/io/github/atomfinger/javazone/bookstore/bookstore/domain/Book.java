package io.github.atomfinger.javazone.bookstore.bookstore.domain;

public class Book {

    private String title, description, author;

    public Book(String title, String description, String author) {
        this.title = title;
        this.description = description;
        this.author = author;
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }

    public String getAuthor() {
        return author;
    }
}
