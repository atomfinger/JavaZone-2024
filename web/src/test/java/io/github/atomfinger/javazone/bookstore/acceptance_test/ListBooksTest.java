package io.github.atomfinger.javazone.bookstore.acceptance_test;

import io.github.atomfinger.javazone.bookstore.bookstore.domain.Book;
import org.approvaltests.Approvals;
import org.approvaltests.JsonApprovals;
import org.approvaltests.core.Options;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.web.client.TestRestTemplate;
import org.springframework.boot.test.web.server.LocalServerPort;
import org.springframework.util.Assert;

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
public class ListBooksTest {

    @LocalServerPort
    private int port;

    @Autowired
    private TestRestTemplate restTemplate;

    @Test
    void greetingShouldReturnDefaultMessage() {
        var result = restTemplate.getForObject("http://localhost:" + port + "/api/books", String.class);
        Approvals.verify(result);
    }
}
