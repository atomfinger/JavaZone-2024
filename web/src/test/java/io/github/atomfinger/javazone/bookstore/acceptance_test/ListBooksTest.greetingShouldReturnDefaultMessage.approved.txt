[{"title":"Test","description":"More test","author":"John"}]